queue/lms_interface.py

```
@@ -78,8 +78,8 @@ def submit(request):
                                         queue_name=queue_name,
                                         xqueue_header=xqueue_header,
                                         xqueue_body=xqueue_body,
-                                        urls=urls_json,
-                                        keys=keys_json)
+                                        s3_urls=urls_json,
+                                        s3_keys=keys_json)
```

How did this even work with the wrong names?

---

Maybe it didn't? I think it was just silently ignoring those keyword arguments until 1.10 started saying "sorry, those aren't actual field names you know".

---

It appears this was written as part of the Swift work.  You can see that one commit in https://github.com/edx/xqueue/pull/103 names it correctly in the model, but the field names are switched back later in the PR (but not the code that calls it).  The only "benefit" is that this code isn't in production, since the prod AMI builds from the release branch which is hugely behind the master branch.  https://github.com/edx/xqueue/tree/release  Not running master is a known issue, since when we tried to go to master, a bunch of external graders broke (probably related to this inability to pass in the download URLs honestly, although most entries in the submission table don't use that field)  details - https://openedx.atlassian.net/browse/OPS-1880